packages/@uppy/core/src/Uppy.ts

```diff
@@ -1942,10 +1940,7 @@ export class Uppy<M extends Meta, B extends Body> {
 
     const uploadID = nanoid()
 
-    this.emit('upload', {
-      id: uploadID,
-      fileIDs,
-    })
+    this.emit('upload', uploadID, this.getFiles())
```

Comment:
I don't think this is correct. `this.getFiles()` (files in state) is not always equivalent to `fileIDs` (which is an argument to this method)